### ISSUE TITLE
Error while  inserting query

### ISSUE THREAD

**Issue by ehtelahi** [emmavz/quillmates#23]
I just replaced exec_query with exec_cmd and found this error:
![Capture](https://user-images.githubusercontent.com/44977977/173890677-ba056226-823c-4062-9591-d9b57abbae2f.PNG)

**Comment by kevingzhang** [comment]
In order to communicate effecitively,
1. Please check in your code. If you do not want to break the build, check in to a branch. and let me know the name of the branch
2. Write a test case in the test.http. Let me know the name of the request, such as POST http://localhost:3000/api/recommendations.  Make sure if I run this test case, the error will appear. 

By doing so I can reproduce the error and fix. Otherwise it is hard for me to understand what the situation you are facing

Thanks

@ehtelahi ,
and @emmavz  too.

**Comment by ehtelahi** [comment]
The main thing is I have zero knowledge about rust. So It will be great If you at least write a function that can insert a record into the database. I know you already have that code in main.rs but that is only to run test.http queries and it's basically running in a for loop that I don't need. The above error that is showing because of invalid return type and I dont know which return type should I use. I am going to push changes to main branch so you can take a look at save_article example.
Plus I need that function to handle form data request because I need to send an image via ajax. It doesn't allow me to do that. Seems like you are forcing a condition. Plus I dont know how to save an image in rust. I downloaded a library but that producing an error.
Moreover I need to run another sql query in that save_article example that will store some data in notifications table.

**Comment by ehtelahi** [comment]
I have pushed code. Please see save_article_inner function where I have also wrote comments

**Comment by kevingzhang** [comment]
I just commited and make it compile pass. But you did not tell me how to test, so I cannot test it. Please advice or you can test on your own.

It is better than before since you have commit your change so that I can start to help you. If you do not commit your code, there is nothing I can do to help you. But there is one more thing you can make it even better : You either write a unit test or tell me how to test in the UI. If the new written code cannot be run, there is no way to test. So I just make it compiled, but I do not know if it works as expected.



Regarding other questions:
1. Never save images or large text into the database. In the blockchain world, the data storage in the state machine is super expensive. The right way to do is to save it to IPFS or any network-accessible online storage. In this PoC project, it would be easier just store a URL to an image instead the image itself. So you will only allow user to embed a picture that a URL to load. This is good enough. In the real life, if the user draw a picture or take a camera photo, the images will be saved to IPFS. The URL to this image in IPFS will be stored to the database. The URL is much smaller than the image data.
2. Rust is not an easy language. But you do not need to know everything about it. If you got trouble like this , please do as you did this time. (1) Commit code(if you do not want to break the build, commit to a new branch) (2)Write a unit test or a UI that can trigger the error. Let me know how to trigger the error and expectation (for example, make it compile without error is an expectation). (3) assign this issue to me. I will read the description to do as you expect.  If you do not make it as clear as I asked above, it would be hard for me to understand what you want and how to help you.

Since you do not need store image in database,  please remove those image-related libraries. It makes the app unnecessary large.

**Comment by ehtelahi** [comment]
Hi!
1) I know we dont store images directly into the database instead we save the path of that image in the database. Maybe you misunderstood.
Currently we want to store image in some folder on our website a temporary purpose. later on, we will save it to the IPFS. I already discussed it with emma.
so basically I need a function that can simply store user uploaded image to some folder and save that path into our database.
and also I will tell you how you can do ui test of it.

**Comment by kevingzhang** [comment]
> Hi!
> 
> 1) I know we dont store images directly into the database instead we save the path of that image in the database. Maybe you misunderstood.
> 
> Currently we want to store image in some folder on our website a temporary purpose. later on, we will save it to the IPFS. I already discussed it with emma.
>

Can we make it simpler just allow the user to use an image that already in the internet? In this case, just store the URL. Storing images is not the key features of this MVP. User can upload his picture to anywhere online, and copy paste the URL.

@emmavz Do you agree this simplified solution?
 
> so basically I need a function that can simply store user uploaded image to some folder and save that path into our database.
> 

You can use File library to store a bytes buffer to file system. but this is not recommanded. In our real system, you have no access to the hard disk. Actually there is no hard disk in our TEA Project at all. 

Without saving to local hard disk, you can still upload the buffer to IPFS. You can do so using node.js IPFS API. But again, I would suggest not focus on this feature, use URL instead. @emmavz if you agree.

> and also I will tell you how you can do ui test of it.

YOu can test it first, if it is not working, then do as I adviced to make a new issue about it, and I am happy to help

**Comment by ehtelahi** [comment]
@kevingzhang!
can you please check discord? communication will be much more easier. I need quick help.

**Comment by emmavz** [comment]
@kevingzhang 

Yes, the simplified solution would be fine; users can just link their pictures to an existing one on the web for this MVP
